src/NServiceBus.Core/Pipeline/Outgoing/RoutingToDispatchConnector.cs

```diff
@@ -44,15 +44,25 @@ public override Task Invoke(IRoutingContext context, Func<IDispatchContext, Task
         static void LogOutgoingOperations(TransportOperation[] operations)
         {
             var sb = new StringBuilder();
+
             foreach (var operation in operations)
             {
                 if (operation.AddressTag is UnicastAddressTag unicastAddressTag)
                 {
-                    sb.AppendFormat("Destination: {0}" + Environment.NewLine, unicastAddressTag.Destination);
+                    sb.AppendLine($"Destination: {unicastAddressTag.Destination}");
+                }
+
+                sb.AppendLine("Message headers:");
+
+                foreach (var kvp in operation.Message.Headers)
+                {
+                    sb.Append($"{kvp.Key}:{kvp.Value}, ");
                 }
 
-                sb.AppendFormat("Message headers:" + Environment.NewLine + "{0}", string.Join(", ", operation.Message.Headers.Select(h => h.Key + ":" + h.Value).ToArray()));
+                sb.Length -= 2;
+                sb.AppendLine();
```

Comment:
I believe this is actually adding a newline that wasn't in the original version of the code.